pandas/tests/groupby/test_api_consistency.py

```diff
@@ -0,0 +1,136 @@
+"""
+Test the consistency of the groupby API, both internally and with other pandas objects.
+"""
+
+import inspect
+
+import pytest
+
+from pandas import (
+    DataFrame,
+    Series,
+)
+from pandas.core.groupby.generic import (
+    DataFrameGroupBy,
+    SeriesGroupBy,
+)
+
+
+def test_frame_consistency(request, groupby_func):
```

Comment:
Side note: Doing something like this for the entire API would be really cool. It would help the ensure when adding/modifying/deprecating methods & their signatures that consistency remains.